rules/falco_rules.yaml

```diff
@@ -3238,3 +3238,10 @@
     command=%proc.cmdline pid=%proc.pid file=%fd.name parent=%proc.pname gparent=%proc.aname[2] ggparent=%proc.aname[3] gggparent=%proc.aname[4] container_id=%container.id image=%container.image.repository)
   priority: WARNING
   tags: [filesystem, mitre_credential_access, mitre_discovery]
+
+- rule: PTRACE attached to process
+  desc: "This rule detects an attempt to inject code into a process using PTRACE."
+  condition: evt.type=ptrace and evt.dir=> and evt.arg.request=11 and proc_name_exists
```

Comment:
Nice idea @Brucedh! 

Just wanted to add something to this. Code injection is usually performed by issuing the `ptrace` syscall and using the following three requests in my opinion: `PTRACE_ATTACH`, `PTRACE_POKETEXT` and `PTRACE_SETREGS`. The poke text requests usually copy a shell code to the memory of the victim process and the set regs request usually modifies the instruction pointer to execute from that shell code. As you were saying, the attach requests could possibly lead to false positives, so I think we can use the two other requests to improve this rule. 

WDYT?

Reply:
Hey, thanks for the feedback. Do you know why Falco thinks that "POKETEXT" is evt.arg.request=5 and not 4, same with SETREGS being 27 from Falco's point of view but 13 from a C programmer's perspective? https://elixir.bootlin.com/linux/v4.8/source/arch/arm/include/uapi/asm/ptrace.h#L16

Reply:
Hello @nicklangsysdig, the reason is that Falco maps those flags to its internal representations. This is done for portability so that regardless of the system Falco is running, the flags will all have a unique representation. This is useful if flags can change from one OS to another. If you want to dig deeper and know about how Falco redefines these flags, please check it here: https://github.com/falcosecurity/libs/blob/master/driver/ppm_events_public.h#L416-L454
keep in mind that the same reasoning applies to all flags, so this is an advice you can re-use in the future 🙂

Reply:
Thank you for your feedback @loresuso !
I agree it’s a good idea to cover also PTRACE_POKETEXT and PTRACE_SETREGS. I added them along with PTRACE_POKEDATA, which is equivalent to PTRACE_POKETEXT

Reply:
It makes sense, thank you!